providers/azure/aci.go

```diff
@@ -193,6 +197,37 @@ func (p *ACIProvider) GetPod(namespace, name string) (*v1.Pod, error) {
 	return containerGroupToPod(cg)
 }
 
+// GetPodLogs returns the logs of a pod by name that is running inside ACI.
+func (p *ACIProvider) GetContainerLogs(namespace, podName, containerName string) (string, error) {
+	logContent := ""
+	cg, err := p.aciClient.GetContainerGroup(p.resourceGroup, fmt.Sprintf("%s-%s", namespace, podName))
+	if err != nil {
+		// Trap error for 404 and return gracefully
+		if strings.Contains(err.Error(), "ResourceNotFound") {
+			return logContent, nil
+		}
+		return logContent, err
+	}
+
+	if cg.Tags["NodeName"] != p.nodeName {
```

Comment:
I'm not sure what the k8s API Server expects here, but we may want to consider handing back an error, like a 404 or something. Getting empty logs could be misleading.

Reply:
done